pkg/cmd/util/tokencmd/request_token.go

```diff
@@ -205,11 +293,44 @@ func oauthAuthorizeResult(location string) (string, error) {
 		return "", err
 	}
 
-	if accessToken := fragmentValues.Get("access_token"); len(accessToken) > 0 {
-		return accessToken, nil
+	return fragmentValues.Get("access_token"), nil
+}
+
+// oauthCodeFlow performs the OAuth code flow if location has a code parameter.
+// It only returns an error if something "impossible" happens (location is not a valid URL)
+// or a definite OAuth error is encountered during the code flow.  Other errors are assumed to
+// be caused by location not being part of the OAuth flow (it was a redirect that the client
+// needs to follow as part of the challenge flow and not a redirect step in the OAuth flow).
+func oauthCodeFlow(client *osincli.Client, authorizeRequest *osincli.AuthorizeRequest, location string) (string, error) {
+	// Make a request out of the URL since that is what AuthorizeRequest.HandleRequest expects to extract data from
+	req, err := http.NewRequest("GET", location, nil)
+	if err != nil {
+		return "", err
+	}
+
+	authorizeData, err := authorizeRequest.HandleRequest(req)
```

Comment:
Where are we checking if the URL even has a code parameter? Remember that this redirect could be a redirect to an authenticating proxy that we need to follow

Reply:
Right, and this is handled by `errIfOAuthError` which turns all non-`osincli.Error` into `nil`.  `osincli` only ever returns an `osincli.Error` when the URL has an error parameter.  All other errors (such as missing code), are returned as `errors.New` or the like.  Note that the GSSAPI tests passed with this PR, so this change at least works with that auth proxy configuration.

Reply:
Descending into the code flow when we don’t actually have a code param, then ignoring the error seems backwards. It’s simple to check if we have an error or code param first